api/src/OpenTelemetry/Trace/Core.hs

```diff
@@ -394,7 +394,7 @@ addAttribute :: (MonadIO m, A.ToAttribute a)
   -> a 
   -- ^ Attribute value
   -> m ()
-addAttribute (Span s) k v = liftIO $ modifyIORef s $ \i -> i
+addAttribute (Span s) k v = liftIO $ modifyIORef' s $ \i -> i `seq` i
```

Comment:
This `seq` is redundant because `i` comes from the `IORef`, and now that we use `modifyIORef'`, it is already evaluated.

```suggestion
addAttribute (Span s) k v = liftIO $ modifyIORef' s $ \i -> i
```

The other `seq` in this file are also redundant.